src/AppsCompiler.ts

```diff
@@ -143,30 +144,44 @@ export class AppsCompiler {
                 };
 
                 for (const moduleName of moduleNames) {
-                    this.resolver(moduleName, resolvedModules, containingFile, result, this.wd, moduleResHost);
+                    const index = this.resolver(moduleName, resolvedModules, containingFile, result, this.wd, moduleResHost);
+                    if (index === -1) {
+                        modulesNotFound.push({
+                            filename: containingFile,
+                            line: 0,
+                            character: 0,
+                            lineText: '',
+                            message: `Failed to resolve module: ${ moduleName }`,
+                            originalMessage: `Module not found: ${ moduleName }`,
+                            originalDiagnostic: undefined,
+                        });
+                    }
                 }
 
-                // @TODO deal with this later
-                // if (moduleNames.length > resolvedModules.length) {
-                //     const failedCount = moduleNames.length - resolvedModules.length;
-                //     console.log(`Failed to resolved ${ failedCount } modules for ${ info.name } v${ info.version }!`);
-                // }
-
                 return resolvedModules;
             },
         } as LanguageServiceHost;
 
         const languageService = this.ts.createLanguageService(host, this.ts.createDocumentRegistry());
 
-        const coDiag = languageService.getCompilerOptionsDiagnostics();
-        if (coDiag.length !== 0) {
-            console.log(coDiag);
+        try {
+            const coDiag = languageService.getCompilerOptionsDiagnostics();
+            if (coDiag.length !== 0) {
+                console.log(coDiag);
+
+                console.error('A VERY UNEXPECTED ERROR HAPPENED THAT SHOULD NOT!');
+                // console.error('Please report this error with a screenshot of the logs. ' +
+                //     `Also, please email a copy of the App being installed/updated: ${ info.name } v${ info.version } (${ info.id })`);
 
-            console.error('A VERY UNEXPECTED ERROR HAPPENED THAT SHOULD NOT!');
-            // console.error('Please report this error with a screenshot of the logs. ' +
-            //     `Also, please email a copy of the App being installed/updated: ${ info.name } v${ info.version } (${ info.id })`);
+                throw new Error(`Language Service's Compiler Options Diagnostics contains ${ coDiag.length } diagnostics.`);
+            }
+        } catch (e) {
+            if (e.message === 'Debug Failure. False expression.') {
```

Comment:
Maybe I should change this to be `if (modulesNotFound.length !== 0) {` and throw regularly when something else is wrong